shapeshift/web
# fix: account id toggles being shared across different devices

## Description

Fixes an issue where having the same seed initialized in multiple different devices causes account ID toggles to appear flakey and partially working, sometimes requiring the user to toggle and account 2 times. Since account ID toggles were stored at the top level, toggling an account "on" in one device will also cause to toggling "on" when connected to the other. In this state, the account appears "off", but toggling it "on" would actually turn it "off" - so it appears as a dead-ish click. Toggling it a second time works because the state would be aligned with what's displayed.

The solution is to separate account ID toggles by wallet ID, such that a toggle for a given account ID cannot be shared between devices.

## Pull Request Type


## Issue (if applicable)

closes #6928

## Risk
> High Risk PRs Require 2 approvals

Moderate risk - this PR includes changes to the portfolio slice, but the changes are isolated to account management which is toggled off in production.

> What protocols, transaction types or contract interactions might be affected by this PR?

## Testing

Buckle up. The following steps need to be follow precisely as it's very state dependent. If you mess up, clear cache and start over.

1. Start with cleared app cache and 2 different wallets initialized with the same seed. In my testing I used native and a ledger.
2. Open the app on native
3. Open account management and import accounts for a chain - I used ethereum. Note the account number of the next empty account. This will be used to create the broken toggle state as it's currently toggled off.
4. Toggle the account "on", click "done"
5. Switch to the other wallet with the same seed
6. Open account management and import accounts for the same chain you did previously. The account you toggled "on" previously should be "off" as the portfolio state is stored against the wallet ID (this is normal)
7. Toggle the same account "on" as you did previously and click "done"
8. Note in prod the list of accounts did not update - the toggle on did nothing. This PR should fix it - the account import will work without a second attempt
9. In prod, attempt to import the account again. This time it will work.

### Engineering



### Operations



## Screenshots (if applicable) 

Before fix:

https://github.com/shapeshift/web/assets/125113430/6bd44184-4264-4d00-8119-72eabb9556a2

After fix:

https://github.com/shapeshift/web/assets/125113430/0a00f259-8de3-4fcd-932d-a62575e2b92e
